### ISSUE TITLE
[Bug]: File uploading is not working in Firefox if file dialog is opened in iframe

### ISSUE THREAD

**Issue by emakhachek** [microsoft/playwright#8016]
### Playwright version

1.13.0

### Operating system

MacOS

### What browsers are you seeing the problem on?

Firefox

### Other information

Node.js v14.15.3

### What happened? / Describe the bug

Hello!
I’m having trouble with the file uploading. The following snippet works OK in Chromium, but fails in the Firefox.
Filechooser window appears in FF (Juggler) by clicking on browseForFilesLink if page.waitForEvent('filechooser') is removed.
Filechooser window does not appear if page.waitForEvent('filechooser') exists.
Unfortunately, I cannot share a website link since it's private.

### Code snippet to reproduce your bug

```shell
const frame: Frame = await this.getFrame();
const [fileChooser] = await Promise.all([
    page.waitForEvent('filechooser'),
    frame.click(this.browseForFilesLink),
 ]);
await fileChooser.setFiles(path);
```

### Relevant log output

```shell
● FIREFOX_ItemImport_ › Item_is_imported_successfully_and_found_via_search

    page.waitForEvent: Timeout while waiting for event "filechooser"
    =========================== logs ===========================
    waiting for event "filechooser"
    ============================================================

      220 |     async uploadFile(path: string): Promise<void> {
      221 |         const [fileChooser] = await Promise.all([
    > 222 |             page.waitForEvent('filechooser'),
          |                  ^
      223 |             this.waitAndClick(this.browseForFilesLink),
      224 |         ]);
      225 |         await fileChooser.setFiles(path);

      at GeneralPopup.<anonymous> (src/pages/generalPopup.ts:222:18)
      at step (src/pages/generalPopup.ts:48:23)
      at Object.next (src/pages/generalPopup.ts:29:53)
      at src/pages/generalPopup.ts:23:71
      at Object.<anonymous>.__awaiter (src/pages/generalPopup.ts:19:12)
      at GeneralPopup.Object.<anonymous>.GeneralPopup.uploadFile (src/pages/generalPopup.ts:489:16)
      at src/Tests/ItemSourcing/ItemImport.test.ts:42:28
      at step (src/Tests/ItemSourcing/ItemImport.test.ts:33:23)
      at Object.next (src/Tests/ItemSourcing/ItemImport.test.ts:14:53)
      at fulfilled (src/Tests/ItemSourcing/ItemImport.test.ts:5:58)
```

**Comment by aslushnikov** [comment]
@emakhachek any chance you can build a small repro for us that we can run locally? It's hard to fix this unless we can debug this!

**Comment by emakhachek** [comment]
Unfortunately, I cannot share a website link since it has a private access. I found file dialog in a frame on a public web. The following snippet works in Chromium but not in FF:
```
  test.only('test', async () => {
    const path = require("path");
    await page.goto('https://design-system.service.gov.uk/components/file-upload/');
    const el = await page.waitForSelector('iframe#iFrameResizer0');
    const frame = await el.contentFrame();
    const [fileChooser] = await Promise.all([
        page.waitForEvent('filechooser'),
        frame.click('input[type=file]'),
     ]);
    await fileChooser.setFiles(path.join(__dirname, `../../../results/jest-junit-report/TEST-result.xml`));
    await page.waitForTimeout(5000);
  })
```
![image](https://user-images.githubusercontent.com/20345335/130037524-1bac9471-8eaf-4ecb-be30-ed7c2a87301d.png)

**Comment by aslushnikov** [comment]
Thank you, I can repro this. That's a bug

**Comment by aslushnikov** [comment]
@emakhachek the fix landed on tip-of-tree and should be available in our canary right now: `npm i @playwright/test@next`

And will also be published in the future v1.15
